roles/pool/src/lib/mod.rs

```diff
@@ -76,7 +82,7 @@ impl PoolSv2 {
                             // we also shut down in case of error
                         },
                     }
-                    break;
+                    break Ok(());
```

Comment:
I assume `break Ok(())` is gonna be used later on with `stop`?

Reply:
The main goal is enabling this https://github.com/stratum-mining/stratum/pull/1066/commits/7c3c3a193a94931a072450caab324f47b6e7c0e2#diff-e6cd9eb2c6af8d3bc58ec083bfeba29dae45ad94d2f34a7df2bf5374bc8777c1R43

This should make this test (partially) redundant https://github.com/stratum-mining/stratum/blob/main/test/message-generator/test/bad-pool-config-test/bad-pool-config-test.json